lib/bap/bap.mli

```diff
@@ -9066,6 +9072,11 @@ module Std : sig
         [def] in [blk].  *)
     val occurs : t -> after:tid -> tid -> bool
 
+    (** [flatten blk] returns [blk] in flattened form in which
+        let expressions and compound expressions are flattened
+        into sequences of simple definitions. *)
```

Comment:
I think we should elaborate a little bit on what the flattened form means, e.g.,

```suggestion
    (** [flatten blk] translates [blk] into the flattened form.
         In the flattened form, all operations are applied to variables, constants, or unknowns, i.e., 
         the operands could not be compound expressions. E.g.,
         {v
            #10 := 11 * (#9 + 13) - 17
         v}
         is translated to,
         {v
            #11 := #9 + 13
            #12 := 11 * #11 
            #10 := #12 - 17
         v}
         @since 2.5.0 *)
```